File: Find-the-Difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        int[] frequency = new int[26]; // Array to store frequency of each character

        // Increment counts for characters in string `s`
        for (char c : s.toCharArray()) {
            frequency[c - 'a']++;
        }

        // Decrement counts for characters in string `t`
        for (char c : t.toCharArray()) {
            frequency[c - 'a']--;
        }

        // Find the character with non-zero frequency
        for (int i = 0; i < 26; i++) {
            if (frequency[i] != 0) {
                return (char) ('a' + i);
            }
        }

        return 'a'; // Default return value (in case no extra character is found)
    }
}

File: Find-the-Difference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        int f[26]={0};
        for(char c:s){
            ++f[c-'a'];
        }
        for(char c:t){
            --f[c-'a'];
        }
        for(int i=0;i<26;i++){
            if(f[i] != 0)return 'a'+i;
        }
        return 'a';
    }
};